frontend/src/layout/NavMain.tsx

```diff
@@ -0,0 +1,150 @@
+    const { t } = useTranslation();
+
+    return (
+        <nav css={{
+            gridArea: "nav",
+
+            // This is necessary for the `fit-content` column width to correctly
+            // apply.
+            overflow: "hidden",
+
+            [`@media not all and (min-width: ${BREAKPOINT}px)`]: {
+                margin: 16,
+                border: "1px solid #888",
+                ...leafNode && { display: "none" },
+            },
+        }}>
+            <div
+                onClick={() => setNavExpanded(prev => !prev)}
+                css={{
+                    padding: "6px 12px",
+                    fontSize: 18,
+
+                    // This is required to make the parent `div` (a grid item)
+                    // span up to 300px wide. This is basically the only way to
+                    // get this grid item span a percentage with a maximum
+                    // width.
+                    minWidth: 300,
+
+                    [`@media not all and (min-width: ${BREAKPOINT}px)`]: {
+                        cursor: "pointer",
+                        "&:hover": {
+                            backgroundColor: "#eee",
+                        },
+                    },
+                }}
+            >
+                {/* TODO: this icon is not optimal */}
+                <FontAwesomeIcon icon={faSitemap} css={{ marginRight: 8 }} />
+                <b>{t("navigation")}</b>
+            </div>
+            <ul css={{
+                listStyle: "none",
+                margin: 0,
+                padding: 0,
+                borderTop: "2px solid #888",
+
+                [`@media not all and (min-width: ${BREAKPOINT}px)`]: {
+                    borderTop: navExpanded ? "1px solid #888" : "none",
+                    height: navExpanded ? "auto" : 0,
+                    overflow: "hidden",
+                },
+            }}>
+                {items.map((item, i) => {
+                    const baseStyle = { padding: "6px 12px", display: "block" };
+                    const inner = item.active
+                        ? <b css={{ ...baseStyle, backgroundColor: "#ddd" }}>
+                            {item.label}
+                        </b>
+                        : <Link
+                            to={item.link}
+                            css={{
+                                ...baseStyle,
+                                textDecoration: "none",
+                                transition: "background-color 0.1s",
+                                "&:hover": {
+                                    transitionDuration: "0.05s",
+                                    backgroundColor: "#eee",
+                                    textDecoration: "underline",
+                                },
+                            }}
+                        >{item.label}</Link>;
+
+                    return (
+                        <li key={i} css={{ borderBottom: "1px solid #ccc" }}>
```

Comment:
Why is that risky business? Isn't this key just so that React can have a unique key per node?

Reply:
Quoting the [appropriate React docs](https://reactjs.org/docs/lists-and-keys.html):

> We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state. Check out Robin Pokorny’s article for an [in-depth explanation on the negative impacts of using an index as a key](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318). If you choose not to assign an explicit key to list items then React will default to using indexes as keys.
>
> Here is an [in-depth explanation about why keys are necessary](https://reactjs.org/docs/reconciliation.html#recursing-on-children) if you’re interested in learning more.

And the reason is basically that React uses the key to track which items changed between rerenders.

Reply:
Mhhh. I guess we could use `item.link`? But if a bug somewhere leads to two items having the same link, the strange things happen, right? So I don't see a better way right now. And I guess with breadcrumbs not getting too long, it shouldn't make a noticeable difference in performance?

Reply:
For the record: After some more internal discussion we agreed that we shouldn't use the index. `id` is used now, which had to be passed through accordingly.

Resolving.